public/app/features/alerting/unified/components/mute-timings/EditMuteTiming.tsx

```diff
@@ -5,14 +5,19 @@ import { useGetMuteTiming } from 'app/features/alerting/unified/components/mute-
 import { useURLSearchParams } from 'app/features/alerting/unified/hooks/useURLSearchParams';
 
 import { useAlertmanager } from '../../state/AlertmanagerContext';
+import { GRAFANA_RULES_SOURCE_NAME } from '../../utils/datasource';
 import { AlertmanagerPageWrapper } from '../AlertingPageWrapper';
 
 import MuteTimingForm from './MuteTimingForm';
 
 const EditTimingRoute = () => {
   const [queryParams] = useURLSearchParams();
   const { selectedAlertmanager } = useAlertmanager();
-  const name = queryParams.get('muteName')!;
+  // we need to replace all backslashes with double backslashes, otherwise the rule will not be found
+  const name =
+    selectedAlertmanager === GRAFANA_RULES_SOURCE_NAME
+      ? queryParams.get('muteName')!.replace(/\\/g, '\\\\')
+      : queryParams.get('muteName')!;
```

Comment:
I think we should do that only if the `alertingApiServer` flag is set to true

Reply:
Could we extract this to a simple function where we could test and ensure it only happens when necessary?

Reply:
💯